### PR TITLE
package: docker images to match the unified release format

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Rename docker images from the package system to match the expected naming by the [unified release](https://github.com/elastic/beats/blob/1740cbd1353a2af98600f5a6ce8031822de5a6e0/.ci/scripts/prepare-release-manager.sh#L22-L38).

Therefore, the new name will end with either `docker-image-linux-amd64.tar.gz` or `docker-image-linux-arm64.tar.gz`.

## Why is it important?

Simplify the Daily Release Snapshots manipulation -> https://github.com/elastic/apm-server/pull/7530#discussion_r828941426

## Test

If I run `TYPES=docker mage package` then

```
Generated fields.yml for metricbeat to /Users/vmartinez/work/src/github.com/elastic/beats/x-pack/metricbeat/fields.yml
>> Building metricbeat.yml for linux/amd64
>> Building metricbeat.reference.yml for linux/amd64
>> Building metricbeat.docker.yml for linux/amd64
exec: go "list" "-m"
>> buildGoDaemon: Building for linux/arm64
>> golangCrossBuild: Building for linux/arm64
>> buildGoDaemon: Building for linux/386
>> golangCrossBuild: Building for windows/amd64
>> golangCrossBuild: Building for darwin/amd64
>> golangCrossBuild: Building for darwin/arm64
>> golangCrossBuild: Building for linux/amd64
>> golangCrossBuild: Building for windows/386
...
>> Testing package contents
package ran for 6m51.279139761s
```

## Actions

- [x] Fix the UTs in the packaging
- [ ] Look for consumers if any
- [x] Update docs/changelog if needed?
- [ ] Gather feedback from the teams

### CODE REVIEW

**Comment by rdner** [comment]
@v1v is this PR still relevant?

**Comment by jlind23** [comment]
@v1v any update on this?

**Comment by v1v** [comment]
This is still valid

**Comment by mdelapenya** [comment]
@v1v I guess the e2e will need a change here: https://github.com/elastic/e2e-testing/blob/50f964a93042dcb0b551e6fd0385400be8d3217a/pkg/downloads/versions.go#L358-L366

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -10,8 +10,10 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...main[Check the HEAD dif
 
 *Affecting all Beats*
 
+- Rename docker tar.gz packages to docker-image-linux-<arch>.tar.gz. {pull}30895[30895]
 - Update to Go 1.17.9 {issue}31350[31350]
 
+
```
Comment (v1v) [inline CHANGELOG.next.asciidoc:16]:
```suggestion
```

**Comment by v1v** [comment]
I'll close this issue, and let the DC team drive this change if needed in the future

**Pull request closed without merging**